### DISCUSSION TITLE
Questions about budibase's capability (tables editing and file down/uploading)

### DISCUSSION THREAD

**Discussion by La1x** [Budibase/budibase#4907] [General]
Hi!
I want to create an app where users can edit tables in web app or download it in csv/xls files, edit them and upload back.

I'm trying to figureout is budibase's table component support editing values for end users?
Is budibase has component for downloading/uploading files with custom backend (for example with rest from/to my own backend)?

**Comment by mjashanks** [comment]
Hi Sergey

Unfortunately we do not have a feature that allows end-users to upload data. It's definitely something that we want to do in time.

The table component doesn't support direct editing. However, you can open rows inside a modal, and edit them from there - meaning that you don't have to leave the screen (where the table lives).

You can connect Budibase to your own backend, via our REST connector. You can make the Budibase app send data directly to your REST API, our you can setup an automation that copies records from Budibase to your API - e.g. everytime a record is Created/Updated.

**Reply by La1x** [reply]
Hi Michael! Thanks for answer!
